### PR TITLE
New browser chrome and content window implementation

### PR DESCRIPTION
**Pull request by rainemak**
This takes different pipeline into use from xulrunner-qt5 and qtmozembed-qt5.

This is not yet ready but can be already reviewed.

### CODE REVIEW

**Comment by rojkov** [comment]
LGTM given the conflicts are resolved and the commented out dead code is removed.

File: src/declarativewebpage.h [hunk 1]
```diff
@@ -89,23 +84,24 @@ public slots:
     void fullscreenHeightChanged();
     void toolbarHeightChanged();
 
+#if 0
 protected:
     void componentComplete();
+#endif
```
Comment (rainemak) [inline src/declarativewebpage.h]: Should be removed

**Comment by rainemak** [comment]
LGTM

**Comment by rainemak** [comment]
Minor cleanups and optimizations can be done after this one is merged.

**Comment by rojkov** [comment]
LGTM

**Pull request merged**